### ISSUE TITLE
Backpack inventory tracking incorrect

### ISSUE THREAD

**Issue by pwerken** [EDCD/EDMarketConnector#1115]
**Please complete the following information:**
 - Version 5.0.3-beta0 from source (`commit a710f97c29c5333efa32e5d6670da3b29c81f726 (HEAD -> refs/heads/develop, refs/remotes/origin/develop)`
 - Game Version: 'Odyssey'
 - OS: Linux Debian 11.0
 - OS Locale: English

**Describe the bug**
BackPack inventory tracking has problems keeping a correct count, mainly because of overlapping journal events.
There are `BackpackChange` events that add or remove to the count, but there are also more specific events that change the count : `CollectItems`, `DropItems`, `UseConsumable` and `TransferMicroResources` (and maybe more, these are the once).
This causes the inventory count to be modified twice for the same collect/drop/use item action.

What I'm observing is that a `BackpackChange` is often, but not always (! :frowning_face: ), preceded by one of the more specific events.
The exceptions I've noticed thus far are when downloading data or throwing a grenade, those only have a `BackpackChange` event.

**To Reproduce**
I've included the relevant journal entries for each step.
1. Go to location on foot (anywhere but the station concourse)
2. Transfer a item from / to the ship storage
```
{ "timestamp":"2021-05-24T20:33:10Z", "event":"TransferMicroResources", "Transfers":[ { "Name":"bloodtestresults", "Name_Localised":"Blood Test Results", "Category":"Data", "Count":1, "Direction":"ToBackpack" }, { "Name":"chemicalinventory", "Name_Localised":"Chemical Inventory", "Category":"Data", "Count":1, "Direction":"ToBackpack" }, { "Name":"chemicalinventory", "Name_Localised":"Chemical Inventory", "Category":"Data", "Count":1, "Direction":"ToBackpack" }, { "Name":"combattrainingmaterial", "Name_Localised":"Combat Training Material", "Category":"Data", "Count":2, "Direction":"ToBackpack" }, { "Name":"evacuationprotocols", "Name_Localised":"Evacuation Protocols", "Category":"Data", "Count":2, "Direction":"ToBackpack" } ] }
{ "timestamp":"2021-05-24T20:33:10Z", "event":"BackpackChange", "Added":[ { "Name":"bloodtestresults", "Name_Localised":"Blood Test Results", "OwnerID":0, "Count":1, "Type":"Data" } ] }
{ "timestamp":"2021-05-24T20:33:10Z", "event":"BackpackChange", "Added":[ { "Name":"chemicalinventory", "Name_Localised":"Chemical Inventory", "OwnerID":0, "Count":1, "Type":"Data" } ] }
{ "timestamp":"2021-05-24T20:33:10Z", "event":"BackpackChange", "Added":[ { "Name":"chemicalinventory", "Name_Localised":"Chemical Inventory", "OwnerID":4470318, "Count":1, "Type":"Data" } ] }
{ "timestamp":"2021-05-24T20:33:10Z", "event":"BackpackChange", "Added":[ { "Name":"combattrainingmaterial", "Name_Localised":"Combat Training Material", "OwnerID":4470318, "Count":2, "Type":"Data" } ] }
{ "timestamp":"2021-05-24T20:33:10Z", "event":"BackpackChange", "Added":[ { "Name":"evacuationprotocols", "Name_Localised":"Evacuation Protocols", "OwnerID":4470318, "Count":2, "Type":"Data" } ] }
{ "timestamp":"2021-05-24T20:33:39Z", "event":"TransferMicroResources", "Transfers":[ { "Name":"bloodtestresults", "Name_Localised":"Blood Test Results", "Category":"Data", "Count":1, "Direction":"ToShipLocker" }, { "Name":"chemicalinventory", "Name_Localised":"Chemical Inventory", "Category":"Data", "Count":1, "Direction":"ToShipLocker" }, { "Name":"chemicalinventory", "Name_Localised":"Chemical Inventory", "Category":"Data", "Count":1, "Direction":"ToShipLocker" }, { "Name":"combattrainingmaterial", "Name_Localised":"Combat Training Material", "Category":"Data", "Count":2, "Direction":"ToShipLocker" }, { "Name":"evacuationprotocols", "Name_Localised":"Evacuation Protocols", "Category":"Data", "Count":2, "Direction":"ToShipLocker" } ] }
{ "timestamp":"2021-05-24T20:33:39Z", "event":"BackpackChange", "Removed":[ { "Name":"bloodtestresults", "Name_Localised":"Blood Test Results", "OwnerID":0, "Count":1, "Type":"Data" } ] }
{ "timestamp":"2021-05-24T20:33:39Z", "event":"BackpackChange", "Removed":[ { "Name":"chemicalinventory", "Name_Localised":"Chemical Inventory", "OwnerID":0, "Count":1, "Type":"Data" } ] }
{ "timestamp":"2021-05-24T20:33:39Z", "event":"BackpackChange", "Removed":[ { "Name":"chemicalinventory", "Name_Localised":"Chemical Inventory", "OwnerID":4470318, "Count":1, "Type":"Data" } ] }
{ "timestamp":"2021-05-24T20:33:39Z", "event":"BackpackChange", "Removed":[ { "Name":"combattrainingmaterial", "Name_Localised":"Combat Training Material", "OwnerID":4470318, "Count":2, "Type":"Data" } ] }
{ "timestamp":"2021-05-24T20:33:39Z", "event":"BackpackChange", "Removed":[ { "Name":"evacuationprotocols", "Name_Localised":"Evacuation Protocols", "OwnerID":4470318, "Count":2, "Type":"Data" } ] }
```
4. Drop item and pick the item back up:
```
{ "timestamp":"2021-05-25T20:15:42Z", "event":"DropItems", "Name":"memorychip", "Name_Localised":"Memory Chip", "Type":"Component", "OwnerID":0, "Count":6 }
{ "timestamp":"2021-05-25T20:15:42Z", "event":"BackpackChange", "Removed":[ { "Name":"memorychip", "Name_Localised":"Memory Chip", "OwnerID":0, "Count":6, "Type":"Component" } ] }
{ "timestamp":"2021-05-25T20:15:46Z", "event":"CollectItems", "Name":"opticallens", "Name_Localised":"Optical Lens", "Type":"Component", "OwnerID":0, "Count":1, "Stolen":false }
{ "timestamp":"2021-05-25T20:15:46Z", "event":"BackpackChange", "Added":[ { "Name":"opticallens", "Name_Localised":"Optical Lens", "OwnerID":0, "Count":1, "Type":"Component" } ] }
```
5. Downloaded data and used a consumable:
```
{ "timestamp":"2021-05-26T18:02:11Z", "event":"CollectItems", "Name":"electricalwiring", "Name_Localised":"Electrical Wiring", "Type":"Component", "OwnerID":0, "Count":1, "Stolen":false }
{ "timestamp":"2021-05-26T18:02:11Z", "event":"BackpackChange", "Added":[ { "Name":"electricalwiring", "Name_Localised":"Electrical Wiring", "OwnerID":0, "Count":1, "Type":"Component" } ] }
{ "timestamp":"2021-05-26T18:09:21Z", "event":"BackpackChange", "Added":[ { "Name":"atmosphericdata", "Name_Localised":"Atmospheric Data", "OwnerID":0, "Count":1, "Type":"Data" } ] }
{ "timestamp":"2021-05-26T18:09:21Z", "event":"BackpackChange", "Added":[ { "Name":"extractionyielddata", "Name_Localised":"Extraction Yield Data", "OwnerID":0, "Count":1, "Type":"Data" } ] }
{ "timestamp":"2021-05-26T18:09:21Z", "event":"BackpackChange", "Added":[ { "Name":"unionmembership", "Name_Localised":"Union Membership", "OwnerID":0, "Count":1, "Type":"Data" } ] }
{ "timestamp":"2021-05-26T18:09:21Z", "event":"BackpackChange", "Added":[ { "Name":"geologicaldata", "Name_Localised":"Geological Data", "OwnerID":0, "Count":1, "Type":"Data" } ] }
{ "timestamp":"2021-05-26T18:13:36Z", "event":"UseConsumable", "Name":"energycell", "Name_Localised":"Energy Cell", "Type":"Consumable" }
{ "timestamp":"2021-05-26T18:13:36Z", "event":"BackpackChange", "Removed":[ { "Name":"energycell", "Name_Localised":"Energy Cell", "OwnerID":0, "Count":1, "Type":"Consumable" } ] }
```
6. Throw a grenade:
```
{ "timestamp":"2021-05-24T12:43:35Z", "event":"UnderAttack", "Target":"You" }
{ "timestamp":"2021-05-24T12:43:36Z", "event":"BackpackChange", "Removed":[ { "Name":"amm_grenade_frag", "Name_Localised":"Frag Grenade", "OwnerID":0, "Count":1, "Type":"Consumable" } ] }
{ "timestamp":"2021-05-24T12:43:49Z", "event":"UnderAttack", "Target":"You" }
```

**Expected behavior**
Better tracking of the backpack contents.

**Additional context**
monitor.py handles both the `BackpackChange` event and the more specific ones. It's probably best to only process the `BackpackChange` and ignore the others.

I'm aware there are some shortcomings / oddities with the player journal with regards to the backpack inventory.
I've mainly made this bug because I didn't see it in the know issues list.

**Comment by A-UNDERSCORE-D** [comment]
Whoops.

**Comment by Athanasius** [comment]
The events for this have undergone changes in today's patch, 4.0.0.200.

> LoadGame - Added language, gameversion, buildnumber
> Shipyard - Only write prices for ships available to buy
> Backpack - include contents info in the journal file at startup and disembark
> BackpackMaterials - remove redundant event
> TransferMicroResources - change from amount transferred, to LockerOldCount & LockerNewCount
> Suit & Weapon loadout info - add SuitMods, WeaponMods, WeaponClass

So we'll need to review this all.

**Comment by A-UNDERSCORE-D** [comment]
Should be fixed in 5.0.3 -- But as you're running develop you should just need a pull
